module/zstd/lib/common/zstd_internal.h

```diff
@@ -12,6 +12,15 @@
 #ifndef ZSTD_CCOMMON_H_MODULE
 #define ZSTD_CCOMMON_H_MODULE
 
+/*
+ * Disable the aarch64 NEON SIMD intrinsics for kernel builds.  Safely
+ * using them in the kernel context requires saving/restoring the FPU
+ * registers which is not currently done.
+ */
+#ifdef _KERNEL
```

Comment:
This threw me for a sec, until I realised the only uses of `ZSTD_NO_INTRINSICS` is in conjunction with `__ARM_NEON`. I don't think I care a lot, like, this code is already tangled and hard to follow and this is an obvious improvement, but it took a moment so is worth mentioning.

Reply:
Nah, I think this is fine. It's much easier to express in the source file, and at some point when we do tackle an upgrade, it'll show up brightly in the diff so it won't be missed.